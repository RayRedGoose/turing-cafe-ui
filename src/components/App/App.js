import React, { Component } from 'react';
import './App.scss';
import { getReservations, postReservation, deleteReservation } from '../../apiCalls/apiCalls'
import ReservationBoard from '../ReservationBoard/ReservationBoard'
import ReservationForm from '../ReservationForm/ReservationForm'

class App extends Component {
  constructor() {
    super()
    this.state = {
      reservations: null,
      sys: {
        isLoaded: false,
        error: null
      }
    }
  }

  componentDidMount() {
    getReservations()
      .then(reservations => this.setState({reservations, sys: {isLoaded: true}}))
      .catch(error => this.setState({sys: {error}}))
  }

  addNewReserv = (newReserv) => {
    postReservation(newReserv)
      .then(reservation => this.setState({reservations: [...this.state.reservations, reservation]}))
      .catch(error => this.setState({sys: {error}}))
  }

  removeReserv = (id) => {
    deleteReservation(id)
      .then(reservations => this.setState({reservations, sys: {isLoaded: true}}))
      .catch(error => this.setState({sys: {error}}))
  }

  render() {
    return (
      <main className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <ReservationForm addNewReserv={this.addNewReserv} />
        {this.state.sys.isLoaded &&
          <ReservationBoard
            reservations={this.state.reservations}
            removeReserv={this.removeReserv}/>}
      </main>
    )
  }
}

export default App;
