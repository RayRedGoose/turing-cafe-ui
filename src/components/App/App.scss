@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../../assets/cafe-img.jpeg');
  width: 100%;
  height: 100vh;
  overflow: hidden;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
}

.app-title {
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}

button {
  cursor: pointer;
  background-color: #088081;
  border: none;
  color: #fff;
  font-size: 16px;
  padding: 0.4em 2em;
  &:hover {
    background-color: lighten(#088081, 10%);
  }
}
