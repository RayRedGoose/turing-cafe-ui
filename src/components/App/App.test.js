import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe("App", () => {
  let app

  beforeEach(() => {
    app = shallow(<App />)
  })

  it('renders with all data passed in correctlly', () => {
    expect(app).toMatchSnapshot()
  })
})
