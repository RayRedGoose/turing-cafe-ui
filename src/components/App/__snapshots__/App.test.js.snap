// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with all data passed in correctlly 1`] = `
<main
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <ReservationForm
    addNewReserv={[Function]}
  />
</main>
`;
